modules/core/client/app/init.js

```diff
@@ -10,6 +10,20 @@ angular.module(ApplicationConfiguration.applicationModuleName).config(['$locatio
 	}
 ]);
 
+angular.module(ApplicationConfiguration.applicationModuleName).run(function($rootScope, $state, Authentication) {
+    // Check authentication before changing state
+    $rootScope.$on('$stateChangeStart', function(event, toState, toParams, fromState, fromParams) {
+        if (toState.data && toState.data.requiresLogin && Authentication.user === '') {
```

Comment:
@lirantal @rhutchison do you think here we should check the role of Authentication.user? It would make easy for future expansion towards role-based access control. I mean, instead of `data: { requiresLoging: true }`, we could have for example `data: { allowedRoles: ['user'] }`. Should I do that?